sasmodels/models/onion.py

```diff
@@ -345,7 +345,7 @@ def profile(sld_core, radius_core, sld_solvent, n_shells,
         if fabs(A[k]) < 1.0e-16:
             # flat shell
             z.append(z_current + thickness[k])
-            rho.append(sld_out[k])
+            rho.append(sld_in[k])
```

Comment:
Just to be consistent both with the documentation and the C code that uses sld_in. If A=0, the sld of the shell is constant, so normally the user should set sld_out = sld_in and then there are no ambiguities. But if this is not the case, then sld_out is ignored. Using sld_in in this profile function shows the expected flat sld, while using sld_out shows a linear dependence between sld_in and sld_out.